### ISSUE TITLE
Manual install compose dependencies are always mandatory

### ISSUE THREAD

**Issue by theCalcaholic** [nextcloud/all-in-one#4185]
Hi, I tried to start NC AIO with the [manual install](https://github.com/nextcloud/all-in-one/tree/main/manual-install) instructions, however I always got errors like `no such service: <service>` for services that were disabled (because no corresponding profile was active) but were configured as dependency on an active service:

### Steps to reproduce
1. Download, configure and rename the `latest.yml` and `sample.conf` as described in [manual install](https://github.com/nextcloud/all-in-one/tree/main/manual-install).
2. `cd` into the directory and run `docker compose up`

### Expected behavior 

All services without a profile are started (i.e. 'nextcloud-aio-apache', 'nextcloud-aio-database', 'nextcloud-aio-nextcloud', 'nextcloud-aio-notify-push' and 'nextcloud-aio-redis')

### Actual behavior 

```sh
$ docker compose up  
no such service: nextcloud-aio-onlyoffice
$ docker compose create
service nextcloud-aio-talk is required by nextcloud-aio-apache but is disabled. Can be enabled by profiles [talk talk-recording]
```

### Host OS 
Minimal buildroot container (SkiffOS) - I will try to reproduce this on a Fedora machine as well. 

#### Nextcloud AIO version 

Latest main (5cbbe1bab8f5e09af22b1d93ed1d79e662c4b023) / latest tag for containers

#### Current channel 

-

#### Other valuable info 

For reference, when I start the everything with `docker compose --profile collabora --profile talk --profile talk-recording --profile clamav --profile imaginary --profile fulltextsearch --profile onlyoffice up`, the services start up successfully.

**Comment by szaimen** [comment]
Thanks for the report! 
I am wondering why you are the first reporting this since it seems to be a missing feature in compose: https://github.com/compose-spec/compose-spec/pull/404. 
I am not sure thus how we should proceed here. What would be your idea?

**Comment by szaimen** [comment]
Or maybe it is a bug in compose?
![Screenshot_20240206_195641_Brave](https://github.com/nextcloud/all-in-one/assets/42591237/36f38665-92c6-47a3-a67f-20082f124abf)

**Comment by szaimen** [comment]
@Zoey2936 do you have any idea how we could fix this?

**Comment by Zoey2936** [comment]
seems like a bug in compose to me

**Comment by szaimen** [comment]
> seems like a bug in compose to me

Agreed. Please report this to the compose project then @theCalcaholic. Thanks! :)
